openedx/core/djangoapps/waffle_utils/docs/decisions/0004-waffle-util-namespacing.rst

```diff
@@ -0,0 +1,43 @@
+Waffle Util Namespacing
+***********************
+
+Status
+======
+
+Draft
+
+Context
+=======
+
+The toggle classes WaffleFlag and WaffleSwitch rely on several namespace classes (WaffleNamespace, WaffleSwitchNamespace, and WaffleFlagNamespace). In order to create a WaffleFlag (or WaffleSwitch), you must first create a Namespace object.
+
+Other IDAs have active waffle flags and switches that don't use any namespacing, like this `example switch in ecommerce`_. Once WaffleFlag and WaffleSwitch are extracted to be used in other IDAs (see 0002-waffle-utils-extraction), the required namespace class will make this transition more difficult.
+
+Additionally, the fully qualified waffle name, including the namespace, is required in code annotations and the django admin. Since it needs to be manually reconstructed by the developer, it has lead to copy/paste issues that are also difficult to lint.
```

Comment:
Related, it is easier to find the name in code if you can search for the whole name found in admin.